File: spring/server/src/main/java/edu/berkeley/gpporgs/model/Review.java
package edu.berkeley.gpporgs.model;


import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.persistence.Transient;

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "reviews")
public class Review {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private @NonNull Long organizationId;
    private @NonNull Long addressId;
    @Transient
    private @NonNull Address address;
    private @NonNull String region;
    private @NonNull String languageCodes;
    @Transient
    private @NonNull Iterable<String> languages;
    private String difficulties;
    private String sectorIds;
    @Transient
    private @NonNull Iterable<String> sectors;
    private Integer stipend;
    private Integer cost;
    private String duration;
    private @NonNull String workDone;
    private String typicalDay;
    private @NonNull String evaluation;
    private String other;
    private Integer safety;
    private Integer responsiveness;
    private @NonNull Boolean anonymous;
    private @NonNull String reviewerId;
    @Transient
    private User reviewer;
}

File: client/src/app/model/review.ts
import { User } from "./user";
import { Address } from "./address";

export class Review {
  id: number = null;
  created: string = null;
  address: Address = new Address();
  region: number = null;
  languages: string[] = [];
  sectors: number[] = [];
  sectorOther: string = null;
  cost: number = 0;
  stipend: number = 0;
  workDone: string = null;
  evaluation: string = null;
  typicalDay: string = null;
  difficulties: string = null;
  safety: number = 0;
  responsiveness: number = 0;
  duration: string = null;
  other: string = null;
  reviewerId: number = null;
  reviewer: User = null;
  anonymous: boolean = true;
}
